reporters_db/make_csv.py

```diff
@@ -54,7 +54,7 @@ def make_editions_dict(editions):
 
 
 def make_csv():
-    with open("reporters.csv", "w", newline="") as f:
+    with open("reporters.csv", "w", encoding="utf-8", newline="") as f:
```

Comment:
oh ... python 3.10... smart

Reply:
The biggest problem is with windows, specifying encoding ensure cross-platform support when opening files

Reply:
This isn't a python version thing, @flooie. Weirdly — extremely weirdly — when you open a file in Python, it does so using the encoding in the `open()` command. If it's not provided, it uses the default for your OS! We get away with being sloppy because OSX and Linux do the sensible thing and have utf-8 as the default, but Windows is stuck back in cp1252 land, and goes haywire. 

Semgrep will alert when it's missing (I think I added this rule to it).

What shocks me is that they didn't fix this as part of the big py3 push, but I think it was just one bridge too far.

Reply:
I think it’s actually the new standard. Let me find the notes in Python release

Reply:
I see what you mean about the `encoding` parameter not being there on py2, but I think the behavior of using the OS encoding was the same. That is:

```
# py2
open(some-file.txt, 'r')
```

Is the same as:
```
# py3
open(some-file.txt, 'r')
```

I didn't realize `encoding` wasn't a parameter in py2 though. I still think it's weird they didn't just set the default to utf-8. Maybe that's me being an English speaker though.

Reply:
I was thinking about [PEP 597](https://peps.python.org/pep-0597/), which adds a warning for now having an encoding in python 3.10